contest/views.py

```diff
@@ -17,65 +17,235 @@
+                    testcase_contestant_name = submission.user.username
+            contestant_testcase_scoreboard = {'status':testcase_status,'number':testcase_number}
+            testcase_scoreboard[testcase_contestant_name] = {'contestant_scoreboard':contestant_testcase_scoreboard}
+            contestant_testcase_scoreboard = []
+
+    return render(request, 'contest/contest.html',{'contest':contest,'clarification_list':clarification_list,
+        'contestant_list':contestant_list,'contestant_number':contestant_list.__len__(),
+        'scoreboard':scoreboard,'testcase_scoreboard':testcase_scoreboard,
+        'server_time':serverTime,'problem_pass_rate':problem_pass_rate},
+        context_instance = RequestContext(request, processors = [custom_proc]))
 
-    contest1 = {'name':'First contest','start_time':'2014/12/27 15:30:00','end_time':'2014/12/27 16:00:00',
-                    'contest_contestant':200,'contest_owner':'ma in joe','problem_list':problemList,'contestant_list':contestantList
-                    ,'clarification_list':clarificationList}
+def new(request):
+    if request.user.has_judge_auth():
+        if request.method == 'GET':
+            form = ContestForm()
+            return render(request,'contest/editContest.html',{'form':form})
+        if request.method == 'POST':
+            form = ContestForm(request.POST)
+            if form.is_valid():
+                new_contest = form.save()
+                logger.info('Contest: Create a new contest %s!' % new_contest.id)
+                return HttpResponseRedirect('/contest/')
+    else:
+        raise PermissionDenied
+    
 
-    return render(request, 'contest/contest.html',{'contest':contest1,'problem_list':problemList,
-        'contestant_list':contestantList,'server_time':serverTime},
-        context_instance = RequestContext(request, processors = [custom_proc]))
+def edit(request,contest_id):
+    contest = get_object_or_404(Contest,id = contest_id)
+    if user_info.has_c_ownership(request.user,contest):
+        if request.method == 'GET':        
+            contest_dic = model_to_dict(contest)
+            form = ContestForm(initial = contest_dic)
+            return render(request,'contest/editContest.html',{'form':form})
+        if request.method == 'POST':
+            form = ContestForm(request.POST, instance = contest)
+            if form.is_valid():
+                modified_contest = form.save()
+                logger.info('Contest: Modified contest %s!' % modified_contest.id)
+                return HttpResponseRedirect('/contest/')
+    else:
+        raise PermissionDenied
+
+def delete(request,contest_id):
+    try:
+        contest = Contest.objects.get(id = contest_id)
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not delete contest %s! Contest not found!' % contest_id)
+        raise Http404("Contest does not exist, can not delete.")
+        
+    if user_info.has_c_ownership(request.user,contest):
```

Comment:
Only owner can delete contest. has_c_coownership includes owner and coowner.
Please use request.user == contest.owner instead.